### ISSUE TITLE
Unify parsing of additional network objects

### ISSUE THREAD

**Issue by namdre** [DLR-TS/sumo#3863]
Netedit duplicates a lot of code in NLHander

This should be abolished by using an additional abstraction layer just as it is done for sumo and sumo-gui where the same handler creates different object types by using another factory.

**Comment by namdre** [comment]
@palvarezlopez I haven't yet decided how to best distribute the work. This is just a reminder that it needs to be done and it will probably be done by the two of us

**Comment by namdre** [comment]
I agree with the architecture. it's the some thing we use for sharing SUMORouteHandler between MSRouteHandler and RORouteHandler.

There is a small distinction between this architecture and the one used by netload/guinetload:
- MSRouteHandler and RORouteHandler both inherit from SUMORouteHandler
- NLHandler has members of the appropriate type (NLTriggerBuilder vs GUITriggerbuilder)

The inheritance allows stronger decoupling between simulation classes and netedit classes so it works better for this task.

I propose: a common base called SUMOAdditionalHandler for NLHandler and GNEAdditionalHandler

**Comment by palvarezlopez** [comment]
@namdre und why not unify also MSRouteHandler, RORouteHandler, NLTriggerBuilder  and GUITriggerbuilder?

**Comment by namdre** [comment]
- The additional objects parsed by NLTriggerBuilder are not allowed in route files so there is no need for a unification with the route handlers
- There is no duplicate parsing code in NLTriggerBuilder and GUITriggerbuilder (only the object creation is re-implemented in GUITriggerbuilder)
- I would move the NLTriggerBuilder parsing code to the new SUMOAdditionalHandler and then let it call virtual methods to do something with the parsed attributes (NLHandler would use these methods to call the factory functions of it's triggerbuilder (NL or GUI) and GNEAdditionalHandler would use these methods to created the appropriate netedit objects
- I agree that duplicate parsing code for the routehandlers should also be unified: #8387

**Comment by palvarezlopez** [comment]
And unify all route and additional objects in the same handler (CommonXMLParser)?

**Comment by namdre** [comment]
I think having these things separate makes the code more manageable. 
- route handlers never need to parse additionals so they can be more efficient if they don't include code to check them
- additional files may include route-file contents but this is already solved: NLHandler inherits from MSRouteHandler
- in netedit,  I'd actually prefer if the user would need to load an additional file as a route file explicitly to extract the demand-related content

**Comment by palvarezlopez** [comment]
NETEDIT is finished. It continues in 1.11.0

**Comment by palvarezlopez** [comment]
Continue in 1.12.0. Next step is NLHandler
